R/glmCommonFunctions.R

```diff
@@ -0,0 +1,207 @@
+  } else {
+    fullModel <- stats::glm(ff,
+                            family = familyLink,
+                            data = dataset,
+                            weights = get(options$weights))
+    nullModel <- stats::glm(nf,
+                            family = familyLink,
+                            data = dataset,
+                            weights = get(options$weights))
+  }
+
+  glmModels <- list("nullModel" = nullModel,
+                    "fullModel" = fullModel)
+  # combine both models
+  jaspResults[["glmModels"]] <- createJaspState()
+  jaspResults[["glmModels"]]$dependOn(optionsFromObject = jaspResults[["modelSummary"]])
+  jaspResults[["glmModels"]]$object <- glmModels
+
+  return(glmModels)
+
+}
+
+.hasNuisance <- function(options) {
+  return(any(sapply(options$modelTerms, function(x) x[["isNuisance"]])))
+}
+
+.glmCreatePlotPlaceholder <- function(container, index, title) {
+  jaspPlot            <- createJaspPlot(title = title)
+  jaspPlot$status     <- "running"
+  container[[index]]  <- jaspPlot
+  return()
+}
+
+
+.glmStdResidCompute <- function(model, residType, options) {
+  if (residType == "deviance") {
+    stdResid <- rstandard(model)
+  }
+  else if (residType == "Pearson") {
+    phiEst <- summary(model)[["dispersion"]]
+    resid <- resid(model, type="pearson")
+    stdResid <- resid / sqrt(phiEst * (1 - hatvalues(model)))
+  }
+  else if (residType == "quantile") {
+    jaspBase::.setSeedJASP(options)
+    resid <- statmod::qresid(model)
+    stdResid <- resid / sqrt(1 - hatvalues(model))
+  } else {
+    stdResid <- NULL
+  }
+
+  return(stdResid)
+}
+
+.glmInsertPlot <- function(jaspPlot, func, ...) {
+  p <- try(func(...))
+
+  if (inherits(p, "try-error")) {
+    errorMessage <- .extractErrorMessage(p)
+    jaspPlot$setError(gettextf("Plotting is not possible: %s", errorMessage))
+  } else {
+    jaspPlot$plotObject <- p
+    jaspPlot$status     <- "complete"
+  }
+}
+
+.constInfoTransform <- function(family, x) {
+  switch(family,
+         "bernoulli" = 1/(sin(sqrt(x))),
+         "binomial" = 1/(sin(sqrt(x))),
+         "poisson" = sqrt(x),
+         "Gamma" = log(x),
+         "inverse.gaussian" = 1/sqrt(x),
+         "gaussian" = x)
+}
+
+.constInfoTransName <- function(family) {
+  switch(family,
+         "bernoulli" = expression(sin^-1 * sqrt("Fitted values")),
+         "binomial" = expression(sin^-1 * sqrt("Fitted values")),
+         "poisson" = expression(sqrt("Fitted values")),
+         "Gamma" = expression(log("Fitted values")),
+         "inverse.gaussian" = expression(1/sqrt("Fitted Values")),
+         "gaussian" = "Fitted values")
+}
+
+.capitalize <- function(x) {
+  x_new <- paste(toupper(substr(x, 1, 1)), substr(x, 2, nchar(x)), sep="")
+  return(x_new)
+}
+
+# function for multicollineary statistics, taken from the source code of car::vif
+# car version: 3.0-13 see https://rdrr.io/cran/car/src/R/vif.R
+.vif.default <- function(mod, ...) {
```

Comment:
why? If you modified anything in this function, please mention this in a comment. If it's to cut dependencies, also mention it in a comment.

Reply:
It's to remove the dependency. Importing the entire car package would cause JASP to crash when loading the module. Will put that as a comment.